pages/_app.js

```diff
@@ -125,18 +147,48 @@ export default function App({ Component, pageProps }) {
       return;
     }
 
-    const newCategory = { name: capitalizeFirstLetter(category), id: ulid() };
-    setCategories([...categories, newCategory]);
-    setSuccessMessage("Category successfully added!");
+    const response = await fetch("/api/categories", {
+      method: "POST",
+      headers: {
+        "Content-Type": "applications/json",
+      },
+      body: JSON.stringify(correctFormat),
+    });
+
+    if (response.ok) {
+      mutateCategories();
+    }
   }
 
   function handleOpenEditModeCategory(category) {
     setOriginalCategoryName(category.name);
-    setIsEditCategory(category.id);
+    setIsEditCategory(category._id);
     setcategoryToEdit(category);
   }
 
-  function handleSaveEditCategory(editedCategory) {
+  async function handleSaveEditCategory(editedCategory) {
```

Comment:
you could rename this to `handleEditCategory` for consistency.

Reply:
Thanks for the hint, but we choose this naming to be sure what this function do. We allready had handleEditCategory and then we renamed it.

Reply:
I can not follow your reason. 'handleEditCategory' tells perfectly what it does.